File: hotel-app/src/main/java/hotelico/service/booking/application/deal/domain/infrastructure/dto/CustomerDealDTO.java
// path: hotelico/service/booking/application/deal/domain/infrastructure/dto/CustomerDealDTO.java

package hotelico.service.booking.application.deal.domain.infrastructure.dto;

import eu.getsoftware.hotelico.clients.common.domain.domainIDs.CustomerDomainEntityId;
import eu.getsoftware.hotelico.clients.common.domain.domainIDs.HotelDomainEntityId;
import eu.getsoftware.hotelico.clients.common.dto.BasicDTO;
import eu.getsoftware.hotelico.clients.common.utils.AppConfigProperties;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

import java.util.Date;

@Getter
@Setter
@RequiredArgsConstructor
public class CustomerDealDTO implements BasicDTO<CustomerDomainEntityId> {

    private final CustomerDomainEntityId domainEntityId;
    private final boolean active;

    private long activityId = 0L;
    private long consistencyId = 0L;

    private boolean timeValid = true;
    private boolean lastMinute = false;
    private boolean closed = false;

    private HotelDomainEntityId hotelDomainId;
    private CustomerDomainEntityId senderDomainId;

    private int dealDaysDuration = 0;
    private Double totalMoney;

    private Date validFrom;
    private Date validTo;

    private String title = "";
    private String timeString = "";

    private String activityArea = "";
    private String dealCode = "";
    private String tablePosition = "";
    private String dealStatus = "";
    private String shortDescription = "";
    private String description = "";

    private String pictureUrl;
    private String previewPictureUrl;

    public String getValidToString() {
        return validTo != null ? AppConfigProperties.dateFormat.format(validTo) : "";
    }

    public String getValidFromString() {
        return validFrom != null ? AppConfigProperties.dateFormat.format(validFrom) : "";
    }

    public String getPictureUrl() {
        return AppConfigProperties.addHostPrefixOnDemand(pictureUrl);
    }

    public String getPreviewPictureUrl() {
        String picUrl = AppConfigProperties.isEmptyString(previewPictureUrl)
                ? AppConfigProperties.PREVIEW_HOTEL_NOT_AVAILABLE_URL
                : previewPictureUrl;

        return AppConfigProperties.addHostPrefixOnDemand(picUrl);
    }

    public String getTimeString() {
        return consistencyId > 0 ? AppConfigProperties.getTimeFormatted(new Date(consistencyId)) : null;
    }
}

File: hotel-app/src/main/kotlin/eu/getsoftware/hotelico/hotelapp/application/deal/domain/infrastructure/dto/CustomerDealDTO.kt
package eu.getsoftware.hotelico.hotelapp.application.deal.domain.infrastructure.dto;

import eu.getsoftware.hotelico.clients.common.domain.domainIDs.CustomerDomainEntityId
import eu.getsoftware.hotelico.clients.common.domain.domainIDs.HotelDomainEntityId
import eu.getsoftware.hotelico.clients.common.dto.BasicDTO
import eu.getsoftware.hotelico.clients.common.utils.AppConfigProperties
import java.util.*

data class CustomerDealDTO(val initId: Long): BasicDTO<CustomerDomainEntityId>(/*initId*/)
{
    /**
     * owner activity
     */
    var activityId = 0L

    var consistencyId = 0L

    var timeValid = true

    var lastMinute = false

    var closed = false

    var hotelDomainId: HotelDomainEntityId? = null

    var senderDomainId: CustomerDomainEntityId? = null

    var dealDaysDuration = 0

    var totalMoney: Double? = null

    var validFrom: Date? = null

    var validTo: Date? = null

    var title = ""

    private var timeString = ""

    var activityArea = ""

    var dealCode = ""

    var tablePosition = ""

    var dealStatus = ""

    var shortDescription = ""

    var description = ""

    private var pictureUrl: String? = null

    private var previewPictureUrl: String? = null

    fun isActiveBoolean(): Boolean {
        return active
    }

    fun getValidToString(): String {
        return AppConfigProperties.dateFormat.format(validTo)
    }

    fun getValidFromString(): String {
        return AppConfigProperties.dateFormat.format(validFrom)
    }

    fun getPictureUrl(): String? {
        return AppConfigProperties.addHostPrefixOnDemand(pictureUrl)
    }

    fun setPictureUrl(pictureUrl: String) {
        this.pictureUrl = pictureUrl
    }

    fun getPreviewPictureUrl(): String? {
        
        val picUrl = if (AppConfigProperties.isEmptyString(previewPictureUrl)) AppConfigProperties.PREVIEW_HOTEL_NOT_AVAILABLE_URL else previewPictureUrl

        return AppConfigProperties.addHostPrefixOnDemand(picUrl)
    }

    fun setPreviewPictureUrl(previewPictureUrl: String) {
        this.previewPictureUrl = previewPictureUrl
    }

    fun getTimeString(): String? {
        return if (consistencyId > 0) AppConfigProperties.getTimeFormatted(Date(consistencyId)) else null
    }
}
